src/NuGet.Core/NuGet.DependencyResolver.Core/GraphModel/GraphOperations.cs

```diff
@@ -377,8 +377,10 @@ private static void WalkTreeDectectConflicts<TItem>(GraphNode<TItem> node, Confl
                     var acceptedType = LibraryDependencyTargetUtils.Parse(acceptedNode.Item.Key.Type);
                     var childType = childNode.Key.TypeConstraint;
 
+                    // Skip the check if a project reference override a package dependency
                     // Check the type constraints, if there is any overlap check for conflict
-                    if ((childType & acceptedType) != LibraryDependencyTarget.None)
+                    if ((acceptedType & (LibraryDependencyTarget.Project | LibraryDependencyTarget.ExternalProject)) == LibraryDependencyTarget.None
```

Comment:
This is good...but can you please not merge super quickly unless a change has to go in immediately?